src/agni/server/manager/UserManager.java

```diff
@@ -14,10 +16,113 @@ public UserManager(InfoSender infoSender,
         this.infoSender = infoSender;
         this.userDataGuard = userDataGuard;
     }
-
-	@Override
-	public void userRequest(String ip, byte type, String testMessage) {
-		// TODO Auto-generated method stub
-		
+    private boolean friendHasAccepted(String username, String friend) {
+    	String [] friendList = userDataGuard.getFriends(friend); 
+		if(Arrays.asList(friendList).contains(username)) {
+			return true;
+		}
+		else {
+			return false;
+		}
 	}
+    public enum UserRequestType {
+        JOIN_CHAT((byte)0x00),
+        LEAVE_CHAT((byte)0x01),
+        FRIENDS_LIST((byte)0x02),
+        FRIEND_STATUS((byte)0x03),
+        ADD_FRIEND((byte)0x04),
+        LOGOUT((byte)0x05);
+
+        private final byte bytes;
+        private UserRequestType(byte bytes) {
+            this.bytes = bytes;
+        }
+
+        public final byte bytes() {
+            return bytes;
+        }
+    }
+    @Override
+    public void userRequest(String ip, byte type, String argument) {
+        // TODO Auto-generated method stub
+    	String username = userDataGuard.getUsername(ip);
+    	if(type == UserRequestType.JOIN_CHAT.bytes()) { 			// join chat; argument = group to join
+        	String groupName = argument; 
+        	if (userDataGuard.groupExists(groupName)) {
+        		if (!userDataGuard.userCurrentChat(username).equals(null)) {
+        			userDataGuard.removeUserFromChat(username, userDataGuard.userCurrentChat(username));
+        		}
+        		userDataGuard.addUserToChat(username,groupName); 
+        		infoSender.sendInfo(ip, "success: joining " + groupName);
+        	}
+        	else {
+        		infoSender.sendInfo(ip, "failed: " + groupName + " doesn't exist");
+        	}
+        }
+        else if (type == UserRequestType.LEAVE_CHAT.bytes()){ 	// leave chat 
+        	String groupName = userDataGuard.userCurrentChat(username); 
+        	if (!userDataGuard.userCurrentChat(username).equals(null)) {
+        		userDataGuard.removeUserFromChat(username,groupName); 
+        		infoSender.sendInfo(ip, "success: leaving " + groupName);
+        	}
+        	else {
+        		infoSender.sendInfo(ip, "failed: no current chat open");
+        	}
+        	
+        }
+        else if (type == UserRequestType.FRIENDS_LIST.bytes()){ 	// see friends list 
+        	String [] friendList = userDataGuard.getFriends(username); 
+        	String friends = "";
+        	for (int i = 0; i < friendList.length; i++) {
+        		friends.concat(friendList[i] + " " );
+        		if(userDataGuard.isOnline(friendList[i])) {
+        			friends.concat("online" + " \n" );
```

Comment:
they were added together so that one string would be created so that the infoSender would only have to send one string